components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -277,17 +285,20 @@ public static API importApi(String extractedFolderPath, APIDTO importedApiDTO, B
             ApiTypeWrapper apiTypeWrapperWithUpdatedApi = new ApiTypeWrapper(importedApi);
             addThumbnailImage(extractedFolderPath, apiTypeWrapperWithUpdatedApi, apiProvider);
             addDocumentation(extractedFolderPath, apiTypeWrapperWithUpdatedApi, apiProvider);
-            addAPISequences(extractedFolderPath, importedApi, registry);
-            addAPISpecificSequences(extractedFolderPath, registry, importedApi.getId());
             addAPIWsdl(extractedFolderPath, importedApi, apiProvider, registry);
-            addEndpointCertificates(extractedFolderPath, importedApi, apiProvider, tenantId);
             addSOAPToREST(extractedFolderPath, importedApi, registry);
 
-            if (log.isDebugEnabled()) {
-                log.debug("Mutual SSL enabled. Importing client certificates.");
+            if (!isAdvertiseOnlyAPI(importedApiDTO)) {
+                addAPISequences(extractedFolderPath, importedApi, registry);
+                addAPISpecificSequences(extractedFolderPath, registry, importedApi.getId());
+                addEndpointCertificates(extractedFolderPath, importedApi, apiProvider, tenantId);
```

Comment:
Not related to the PR. But these seems directly going to registry :(

Reply:
Shall we track this separately and fix?

Reply:
Thanks a lot for identifying this. Created an issue [1] to keep this tracked and to perform the task.

[1] https://github.com/wso2/product-apim-tooling/issues/675